src/otx/algorithms/common/adapters/mmcv/utils/config_utils.py

```diff
@@ -698,3 +700,243 @@ def get_data_cfg(config: Union[Config, ConfigDict], subset: str = "train") -> Co
     while "dataset" in data_cfg:
         data_cfg = data_cfg.dataset
     return data_cfg
+
+
+class InputSizeManager:
+    """Class for changing input size and getting input size value by checking data pipeline.
+
+    NOTE: "resize", "pad", "crop", "mosaic", "randomaffine", "multiscaleflipaug" and "AutoAugment"
+    are considered at now. If other data pipelines exist, it can work differently than expected.
+
+    Args:
+        data_config (Dict): Data configuration expected to have "train", "val" or "test" data pipeline.
+        base_input_size (Optional[Union[int, List[int], Dict[str, Union[int, List[int]]]]], optional):
+            Default input size. If it's a None, it's estimated based on data pipeline. If it's an integer,
+            it's expected that all data pipeline have (base_input_size x base_input_size) input size.
+            If it's an integer list, all data pipeline have same (base_input_size[0] x base_input_size[1]) input size.
+            If it's dictionary, each data pipeline has specified input size. It should have format as below:
+                {"train" : [w, h], "val" : [w, h], "test" : [w, h]}
+    """
+
+    PIPELINE_TO_CHANGE: Dict[str, List[str]] = {
+        "resize": ["size", "img_scale"],
+        "pad": ["size"],
+        "crop": ["crop_size"],
+        "mosaic": ["img_scale"],
+        "randomaffine": ["border"],
+        "multiscaleflipaug": ["img_scale"],
+    }
+
+    def __init__(
+        self,
+        data_config: Dict,
+        base_input_size: Optional[Union[int, Tuple[int, int], Dict[str, int], Dict[str, Tuple[int, int]]]] = None,
+    ):
+        self._data_config = data_config
+        if isinstance(base_input_size, int):
+            base_input_size = (base_input_size, base_input_size)
+        elif isinstance(base_input_size, dict):
+            for task in base_input_size.keys():
+                if isinstance(base_input_size[task], int):
+                    base_input_size[task] = (base_input_size[task], base_input_size[task])  # type: ignore[assignment]
+            for data_type in ["train", "val", "test"]:
```

Comment:
Making the variable `DATA_TYPE`=["train", "val", "test"] and reusing this variable looks better to me since ["train", "val", "test"] used repeatedly

Reply:
I see. I'll reflect it.